### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by bjoo**
- Fixed CMakeLists.txts to be able to cope with TIPL installed with 'make install' (rather than just downloaded into TIPL_DIR)

- Added nonlinear image registration and basic op to toplevel CMakeLists.txt
Added some tempraries to fix some compiler errors on Summit e.g.:
TIPL-example/cuda/image_nonlinear_registration/nonlinear_registration.cu(133): error: initial value of reference to non-const must be an lvalue
- removed unused variables

Code compiles and runs on Summit (single device for now)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,4 +8,6 @@ add_subdirectory(cpp)
 if( ENABLE_CUDA )
 add_subdirectory(cuda/image_linear_registration)
 add_subdirectory(cuda/image_resample)
+add_subdirectory(cuda/image_nonlinear_registration)
```
Comment (bjoo) [inline CMakeLists.txt:11]: Added subdirs

File: cuda/image_basic_op/CMakeLists.txt [hunk 2]
```diff
@@ -18,11 +18,12 @@ set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -allow-unsupported-compiler --extended
 set(CMAKE_CUDA_RUNTIME_LIBRARY "Static")
 
 target_compile_definitions(tipl_image_basic_op PUBLIC TIPL_USE_CUDA)
-
-if(EXISTS "${TIPL_DIR}")
-  include_directories(tipl_image_basic_op ${TIPL_DIR})
+find_package(TIPL)
```
Comment (bjoo) [inline cuda/image_basic_op/CMakeLists.txt:21]: This way if TIPL_DIR corresponds to ..../lib/cmake/ with TIPLConfig.cmake in it, find_package will succeed. if not we can use TIPL_DIR to point to the toplevel TIPL directory

File: cuda/image_nonlinear_registration/nonlinear_registration.cu [hunk 3]
```diff
@@ -80,15 +80,13 @@ int main(void)
     {
         tipl::time t("  accumulate displacement using cpu:");
         float theta = 0.0;
-        bool terminated = false;
```
Comment (bjoo) [inline cuda/image_nonlinear_registration/nonlinear_registration.cu:83]: GCC flagged terminated as an unused variable here

File: cuda/image_nonlinear_registration/nonlinear_registration.cu [hunk 4]
```diff
@@ -108,12 +106,14 @@ int main(void)
 
     check_dif(dis,ddis);
 
-    std::cout << "\n==Emsumble TEST: nonlinear registration==" << std::endl;
```
Comment (bjoo) [inline cuda/image_nonlinear_registration/nonlinear_registration.cu:111]: Typo

File: cuda/image_nonlinear_registration/nonlinear_registration.cu [hunk 5]
```diff
@@ -108,12 +106,14 @@ int main(void)
...
+    std::cout << "\n==Ensemble TEST: nonlinear registration==" << std::endl;
 
     {
         tipl::time t("  nonlinear registration using cpu:");
         bool terminated = false;
-        dis.swap(decltype(dis)(dis.shape()));
+
```
Comment (bjoo) [inline cuda/image_nonlinear_registration/nonlinear_registration.cu:114]: Here there was an issue that 'reference to non-const' must be an lvalue, whereas calling the constructor as previous makes it a temporary (not an lvalue). This caused a compile fail in GCC

**Comment by frankyeh** [comment]
Thanks for your help!

**Pull request merged**